src/main/xerus/monstercat/api/Covers.kt

```diff
@@ -33,8 +38,23 @@ object Covers {
 		return file.inputStream()
 	}
 	
-	private fun coverCacheFile(coverUrl: String) =
-		coverCacheDir.apply { mkdirs() }.resolve(coverUrl.substringAfterLast('/').replaceIllegalFileChars())
+	/** Returns a larger Image of the cover in the requested size using caching.
+	 * @param size the size of the Image - the downloaded image will be at maximum 2048x2048 */
+	fun getCoverImage(coverUrl: String, size: Int = 2048, invalidate: Boolean = false): Image =
+			getCover(coverUrl, invalidate).use { createImage(it, size) }
+	
+	/** Returns an InputStream to the cover in size 2048x2048, using caching. */
+	private fun getCover(coverUrl: String, invalidate: Boolean = false) : InputStream {
```

Comment:
Please merge this method with the `getThumbnail` method, they are almost equal. DRY!

Reply:
I cannot do that, because both use different caching directories. I'd need to add a new argument "thumbnail: Boolean" and it will end up being one big function with an if(thumbnail) getThumbnail else getCover

Reply:
1. They don't have to use different directories. We could also adjust the filename pattern to include the size.
2. The only other thing that is different is the size of the cover. That could be a parameter.

Yes I can also do this after the merge, but I'd like you to learn to recognise these things by yourself ;)

Reply:
imo we should keep them in separate directories. Makes for easier cleanup by hand when folders start to get huge. I'll add a check, if image size is above 128x128 it's not a thumbnail anymore.

Reply:
fcb4c904537d730101db6cc454c453171c86d351